### ISSUE TITLE
IDirectoryInfoFactory.Wrap loses part of absolute path when used with MockFileSystem

### ISSUE THREAD

**Issue by koyote** [TestableIO/System.IO.Abstractions#1049]
**Describe the bug**
If you provide the MockFileSystem's DirectoryInfo.Wrap method with a DirectoryInfo that has an absolute path, the resulting IDirectoryInfo has the wrong file path.

**To Reproduce**
```cs
using Xunit;
using System.IO.Abstractions;
using System.IO.Abstractions.TestingHelpers;

[Fact]
public void Test()
{
    var fs = new MockFileSystem();
    var directoryInfo = new DirectoryInfo(@"C:\subfolder\file");
    var wrappedDirectoryInfo = fs.DirectoryInfo.Wrap(directoryInfo);

    Assert.Equal(directoryInfo.FullName, wrappedDirectoryInfo.FullName); // wrappedDirectoryInfo returns "C:\file"
}
```

**Expected behaviour**
The test above is expected to pass.

**Actual behaviour**
```
Assert.Equal() Failure
             ↓ (pos 3)
Expected: C:\subfolder\file
Actual:   C:\file
             ↑ (pos 3)
```

**Additional context**
Tested on Windows 10, .Net 7, System.IO.Abstractions version 19.2.69

**Comment by fgreinacher** [comment]
Thanks for reporting! Do you know if this has ever worked?

**Comment by koyote** [comment]
Hi @fgreinacher, I do not, no.
I just recently started using this library and quickly encountered the above bug.

But just looking at the code for DirectoryInfo.Wrap:

```cs
public IDirectoryInfo Wrap(DirectoryInfo directoryInfo)
{
    if (directoryInfo == null)
    {
        return null;
    }

    return new MockDirectoryInfo(mockFileSystem, directoryInfo.Name);
}
```
It looks like it's sending in `directoryInfo.Name`, which is only the dir name, not the full path.
I'd assume changing that to `directoryInfo.FullName` would fix it but I don't quite have time at the moment to verify that; maybe this weekend.

**Comment by vbreuss** [comment]
Please note, that the same issue also applies when wrapping a `FileInfo`. I fixed both issues with #1079.
